spek-ide-plugin/intellij-common/src/main/kotlin/org/spekframework/intellij/util.kt

```diff
@@ -0,0 +1,248 @@
+            val resolved = mainReference.resolve()
+            if (resolved != null && resolved is KtNamedFunction) {
+                val synonym = extractSynonymAnnotation(resolved)
+                if (synonym != null && !synonym.excluded) {
+                    val description = extractDescription(callExpression)
+                    if (description != null) {
+                        val parentCallExpression = getParentCallExpression(callExpression)
+                        val currentPath = "${synonym.prefix} $description".trim()
+                        if (parentCallExpression != null) {
+                            val newBuffer = mutableListOf<String>()
+                            newBuffer.add(currentPath)
+                            newBuffer.addAll(buffer)
+                            return extractPath(parentCallExpression, newBuffer)
+                        } else {
+                            // probably root scope
+                            val ktClassOrObject = getKtClassOrObject(callExpression)
+                            if (ktClassOrObject != null) {
+                                val builder = pathBuilderFromKtClassOrObject(ktClassOrObject)
+                                // TODO: PathBuilder is immutable for some reason :noidea: why I did this :)
+                                if (builder != null) {
+                                    builder.append(currentPath)
+                                    buffer.forEach {
+                                        builder.append(it)
+                                    }
+
+                                    return builder.build()
+                                }
+                            }
+                        }
+                    }
+                }
+
+            }
+        }
+    }
+    return null
+}
+
+private enum class SynonymType {
+    Group,
+    Action,
+    Test
+}
+
+private data class SynonymData(val type: SynonymType,
+                               val prefix: String,
+                               val excluded: Boolean)
+
+private fun extractSynonymAnnotation(function: KtNamedFunction): SynonymData? {
+    val lightMethod = function.toLightMethods().firstOrNull()
+    val annotation = lightMethod?.annotations?.find {
+        SYNONYM_CLASSES.contains(it.qualifiedName)
+    }
+
+    return annotation?.let {
+        val type = annotation.findDeclaredAttributeValue("type")!!
+        val prefix = annotation.findDeclaredAttributeValue("prefix")
+        val excluded = annotation.findDeclaredAttributeValue("excluded")
+
+        val synonymText = type.text
+        val synonymType = if (synonymText.endsWith("SynonymType.Group")) {
+            SynonymType.Group
+        } else if (synonymText.endsWith("SynonymType.Action")) {
+            SynonymType.Action
+        } else if (synonymText.endsWith("SynonymType.Test")) {
+            SynonymType.Test
+        } else {
+            throw IllegalStateException()
```

Comment:
Not sure if it is OK to fail here so badly. What if a test supports different synonyms but the plugin does not?

Reply:
I'll probably just log something like `Unsupported synonym: <synonym>`

Reply:
A reminder in case you forgot this.

Reply:
done